.cascade-code/Chapman.edu/_components/left_nav/left_nav_siblings.vtl

```diff
@@ -0,0 +1,144 @@
+                    #set ( $visibleChild = false )
+                    #if ( $asset.assetType == 'folder')
+                        ## Both the folder and index page must be visible
+                        #set ( $folderVisible = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+                        #set ( $folderIndexPage = $_.locatePage("${asset.path}/index" ) )
+                        #set ( $folderIndexVisible = $folderIndexPage.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+                        
+                        #if ( $folderIndexVisible && $folderVisible )
+                            #set ( $visibleChild = true )
+                        #end
+                    #elseif ( $asset.assetType == 'page' && ($asset.name != 'index') )
+                        #set ( $visibleChild = $asset.metadata.getDynamicField('Hide from navigation').value != "Yes" )
+                    #end
+                    
+                    #if ( $visibleChild )
+                        #set ( $discard = $visibleChildren.add($asset) )
+                    #end
+                #end
+                
+                #if ( $indexPage.path == $currentPage.path )
+                    <li class="active">
+                #else
+                    <li>
+                #end
+                
+                #if ( $visibleChildren.size() > 0 )
+                    <a href="${indexPage.link}" class="s-f-idx">
+                        $siblingTitle
+                        <span class="plus">+</span>
+                    </a>
+                    #buildChildren($visibleChildren)
+                    </li>
+                #else
+                    <a href="${indexPage.link}" class="s-f-idx">
+                        $siblingTitle
+                    </a>
+                    </li>
+                #end
+            #end
+        #end
+    #end
+    </ul>
+#end
+
+## Sets up the children of a sibling, submenus in the left nav
+#macro ( buildChildren $children )
+    <ul class="leftNav-sub">
+        
+        #foreach ( $child in $children )
+            #set ( $childTitle = ${_EscapeTool.xml($child.metadata.displayName)} )
+            
+            ## If it's a visible page that isn't the index
+            #if ( $child.assetType == 'page' && ($child.name != 'index') )
+                <li>
+                    <a href="${child.link}" title="${childTitle}" class="c-p">
+                        <span class="bullet">» </span>
+                        $childTitle
+                    </a>
+                </li>
+            #elseif ( $child.assetType == 'folder' )
+                <li>
+                    <a href="${child.link}/index" title="${childTitle}" class="c-f">
+                        <span class="bullet">» </span>
+                        $childTitle
+                    </a>
+                </li>
+            #end
+        #end
+    </ul>
+#end
```

Comment:
I'll do a full review when I return. In the meantime, I think the code can be reorganized to better separate some of the logic from the output. I'll use this macro as an example.

First, I would organize the macro around the child list item and call it like so: 

```
<ul class="leftNav-sub">
  #foreach ( $child in $children )
    #buildLeftNavChild( $child )
  #end
</ul>
```

Then I find it easier to read when the `buildLeftNavChild` macro is focused on a single "output" template:

```
#macro ( buildLeftNavChild $child )
  ## Data
  #set ( $childTitle = ${_EscapeTool.xml($child.metadata.displayName)} )
  #set ( $childIsVisible = true )

  #if ( $child.assetType == 'page' && ($child.name != 'index') )
    #set ( $linkClass = "c-p" )
    #set ( $linkHref = $child.link )
  #elseif ( $child.assetType == 'folder' )
    #set ( $linkClass = "c-f" )
    #set ( $linkHref = "${child.link}/index" )
  #else
    #set ( $childIsVisible = false )
  #end

  ## Output
  #if ( $childIsVisible )
  <li>
    <a class="${linkClass}" href="${linkHref}" title="${childTitle}">
      <span class="bullet">» </span>
      $childTitle
    </a>
  </li>
  #end
#end
```